source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/TimeAndDateFucntionTest.cpp

```diff
@@ -0,0 +1,207 @@
+#include "stdafx.h"
+#include "CppUnitTest.h"
+#include "TextBlock.h"
+#include <time.h>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using namespace AdaptiveCards;
+using namespace std;
+
+namespace AdaptiveCardsSharedModelUnitTest
+{		
+	TEST_CLASS(UnitTest1)
+	{
+	public:
+		TEST_METHOD(ISO8601TransformTest)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+            blck->SetText("{{DATE(2017-02-13T20:46:30Z, Short)}}");
+			Assert::AreEqual<string>("02/13/2017",  blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(ISO8601TransformTestLong)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z, Long)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("Monday, February 13, 2017", blck->GetText());
+			delete blck;
+		}
+		TEST_METHOD(ISO8601TransformTestDefault)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("02/13/2017", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{a{DATE(2017-02-13T20:46:30Z, Short)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("{a{DATE(2017-02-13T20:46:30Z, Short)}}", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest2)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017-02-13T20:46:30Z, Short)}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("{{DATE(2017-02-13T20:46:30Z, Short)}", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(ISO8601TransformBadInputTest3)
+		{
+			TextBlock* blck = new TextBlock();
+			Assert::IsNotNull(blck);
+			string testString = "{{DATE(2017a02-13T20:46:30Z, Short)}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("{{DATE(2017a02-13T20:46:30Z, Short)}", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(TransformToDateTest1)
+		{
+			TextBlock* blck = new TextBlock();
+			string testString = "{{DATE(2017-02-13T20:46:30Z, Short)}}";
+			blck->SetText(testString);
+			Assert::AreEqual<string>("02/13/2017", blck->GetText());
+			delete blck;
+		}
+
+		TEST_METHOD(TransformToDateTest2)
```

Comment:
>TransformToDateTest2 [](start = 14, length = 20)

TranformToTime?

Reply:
thanks

---
In reply to: [148399549](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148399549) [](ancestors = 148399549)